src/components/CollectionProvider/CollectionProvider.tsx

```diff
@@ -3,19 +3,25 @@ import { Props } from './CollectionProvider.types'
 
 export default class CollectionProvider extends React.PureComponent<Props> {
   componentDidMount() {
-    const { id, collection, onFetchCollection } = this.props
+    const { id, collection } = this.props
     if (id && !collection) {
-      onFetchCollection(id)
+      this.fetchCollection(id)
     }
   }
 
   componentDidUpdate(prevProps: Props) {
-    const { id, onFetchCollection } = this.props
-    if (id && id !== prevProps.id) {
-      onFetchCollection(id)
+    const { id, collection } = this.props
+    if (id && id !== prevProps.id && !collection) {
+      this.fetchCollection(id)
     }
   }
 
+  fetchCollection(id: string) {
+    const { onFetchCollection, onFetchCollectionItems } = this.props
+    onFetchCollection(id)
+    onFetchCollectionItems(id)
```

Comment:
Would it make sense to just add this to the items saga? I think this is better because it's not coupled but it's worth mentioning, if we always want both together we can forego of the double action here

Reply:
sure, i moved it to the sagas